sql/2021/performance/web_vitals_by_rank.sql

```diff
@@ -0,0 +1,161 @@
+#standardSQL
+# Core WebVitals by rank
+
+CREATE TEMP FUNCTION IS_GOOD (good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
+  good / (good + needs_improvement + poor) >= 0.75
+);
+
+CREATE TEMP FUNCTION IS_NI (good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
+  good / (good + needs_improvement + poor) < 0.75 AND
+  poor / (good + needs_improvement + poor) < 0.25
+);
+
+CREATE TEMP FUNCTION IS_POOR (good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
+  poor / (good + needs_improvement + poor) >= 0.25
+);
+
+CREATE TEMP FUNCTION IS_NON_ZERO (good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
+  good + needs_improvement + poor > 0
+);
+
+WITH
+base AS (
+  SELECT
+    date,
+    origin,
+    rank,
+
+    fast_fid,
+    avg_fid,
+    slow_fid,
+
+    fast_lcp,
+    avg_lcp,
+    slow_lcp,
+
+    small_cls,
+    medium_cls,
+    large_cls,
+
+    fast_fcp,
+    avg_fcp,
+    slow_fcp,
+
+    fast_ttfb,
+    avg_ttfb,
+    slow_ttfb
+
+  FROM
+    `chrome-ux-report.materialized.metrics_summary`
+  WHERE
+    date IN ('2021-07-01')
+)
+
+SELECT
+  date,
+  rank_grouping,
+  CASE
+    WHEN rank_grouping = 10000000 THEN 'all'
+    ELSE FORMAT("%'d", rank_grouping)
```

Comment:
I'd suggest doing all of the formatting in the Sheet rather than the SQL. It makes it easier to change the formatting after all of the data is in.
```suggestion
    ELSE CAST(rank_grouping AS STRING)
```